junit5/src/main/java/cz/xtf/junit5/listeners/ProjectCreator.java

```diff
@@ -17,12 +34,29 @@
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
-public class ProjectCreator implements TestExecutionListener {
-    private static final OpenShift openShift = OpenShifts.master();
+public class ProjectCreator implements TestExecutionListener, BeforeAllCallback, AfterAllCallback, PostDiscoveryFilter {
 
     @Override
     public void testPlanExecutionStarted(TestPlan testPlan) {
+        createProject();
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext context) {
+        if (OpenShiftConfig.useUniqueNamespacePerTestcase()) {
```

Comment:
Is it possible to get rid of `if (OpenShiftConfig.useUniqueNamespacePerTestcase()) {` ? Could it be somehow unified and handle `OpenShiftConfig.useUniqueNamespacePerTestcase() == false` as a simpler case of `OpenShiftConfig.useUniqueNamespacePerTestcase() == true` ?